File: Selenium/Activities/Activity21.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;
import java.util.Set;

public class Activity21 {
    public static void main(String[] args) {
        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(20));

        driver.get("https://v1.training-support.net/selenium/tab-opener");
        String title = driver.getTitle();
        String parentHandle = driver.getWindowHandle();
        System.out.println("1st window title is " + title);
        System.out.println(parentHandle);

        driver.findElement(By.id("launcher")).click();

        wait.until(ExpectedConditions.numberOfWindowsToBe(2));

        System.out.println("Name of handles are below : " + driver.getWindowHandles());

        for(String handle : driver.getWindowHandles()) {
            driver.switchTo().window(handle);
            }

        wait.until(ExpectedConditions.elementToBeClickable(By.id("actionButton")));
        System.out.println("2nd Window Title is " + driver.getTitle());
        System.out.println(driver.findElement(By.className("content")).getText());

        driver.findElement(By.id("actionButton")).click();
        wait.until(ExpectedConditions.numberOfWindowsToBe(3));

        for(String handle : driver.getWindowHandles()) {
            driver.switchTo().window(handle);
        }

        wait.until(ExpectedConditions.titleIs("Newtab2"));
        System.out.println("3rd Window Title is " + driver.getTitle());
        System.out.println(driver.findElement(By.xpath("/html/body/div[1]/div/div/div/div")).getText());

        driver.quit();

    }
}

File: Selenium/Activity21.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Chrome()
wait = WebDriverWait(driver, 20)

driver.get("https://v1.training-support.net/selenium/tab-opener")
print("1st Window Title is " + driver.title)

driver.find_element(By.ID, "launcher").click()

wait.until(EC.new_window_is_opened)

print(f'Name of handles are below : {driver.window_handles}')

for i in driver.window_handles:
    driver.switch_to.window(i)

wait.until(EC.element_to_be_clickable(driver.find_element(By.ID, "actionButton")))
print("2nd Window Title is " + driver.title)
print(driver.find_element(By.CLASS_NAME, "content").text)

driver.find_element(By.ID, "actionButton").click()
wait.until(EC.new_window_is_opened)

for i in driver.window_handles:
    driver.switch_to.window(i)

wait.until(EC.title_is("Newtab2"))
print("3rd Window Title is " + driver.title)
print(driver.find_element(By.XPATH, "/html/body/div[1]/div/div/div/div").text)


driver.close
